src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -69,6 +70,21 @@ private set
             }
         }
 
+        /// <summary>
+        /// True if all of the following conditions are met: 
+        /// 
+        /// 1. KindSupportsFileRename is true
+        /// 2. Type is not partial OR partial type only has single declaration location
+        /// 3. Type is the only type in the file OR type already matches the file name
+        /// </summary>
+        public bool AllowFileRename { get; }
+
+        /// <summary>
+        /// True if the selected node kind supports renaming at all. This allows the UI to hint
+        /// that the type could support renaming, whether or not the current conditions allow it. 
+        /// </summary>
+        public bool KindSupportsFileRename { get; }
```

Comment:
it feels like tehre are only 3 actual values between these two booleans.  consdier an enum.